### PR TITLE
Refactor Maker unit tests

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR refactors Maker unit tests including the test files `MakerBoardTest.js` and `VirtualBoardTest.js`. This is a follow up to [the PR](https://github.com/code-dot-org/code-dot-org/pull/51068#discussion_r1157694927) that adds a virtual micro:bit board so that users can run an App Lab project for which micro:bit is enabled without a connected device. 

I renamed `MakerBoardTest.js` to `MakerBoardInterfaceTestUtil.js` to better indicate that this file exports a set of tests that verify that a given board properly implements the MakerBoard interface. Thus, I removed references to specific board implementations. Currently, there are four of them - `MicroBitBoard`, `CircuitPlayground`, `VirtualMBBoard`, and `VirtualCPBoard`. These tests should be run without needing to check which specific implementation of the MakerBoard interface is being tested.

I also separated the tests in `VirtualBoardTest.js` into 2 different test files: `VirtualMBBoardTest.js` and `VirtualCPBoardTest.js` mirroring the 2 other existing MakerBoard test files: `MicroBitBoardTest.js` and `CircuitPlaygroundBoardTest.js`.

The last change was refactoring `CircuitPlaygroundBoardTest.js` and `MicroBitBoardTest.js`. 
`itMakesCircuitPlaygroundComponentsAvailable` is a function that runs a set of tests that verify that the Circuit Playground board components and component constructors are available from the interpreter. The function is currently defined in and exported from `CircuitPlaygroundBoardTest.js`. It is called in the same file, but also called in `VirtualCPBoardTest.js` which seems somewhat asymmetric. I moved this function to `CircuitPlaygroundComponentTestUtil.js` where it is now imported in `CircuitPlaygroundBoardTest.js` and `VirtualCPBoardTest.js`. 

Thus, there is greater clarity that `CircuitPlaygroundBoard` and `VirtualCPBoard` both properly implement the`MakerBoard` interface and both make available to the interpreter the board components and constructors.

Similar treatment was done for `itMakesMicroBitComponentsAvailable`.

## Links
[jira ticket](https://codedotorg.atlassian.net/browse/SL-779)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/lib/kits/maker/boards/MakerBoardInterfaceTestUtil.js [hunk 1]
```diff
@@ -2,10 +2,6 @@
 import sinon from 'sinon';
 import {EventEmitter} from 'events'; // see node-libs-browser
 import {expect} from '../../../../../util/reconfiguredChai';
-import CircuitPlaygroundBoard from '@cdo/apps/lib/kits/maker/boards/circuitPlayground/CircuitPlaygroundBoard';
-import VirtualCPBoard from '@cdo/apps/lib/kits/maker/boards/VirtualCPBoard';
-import MicroBitBoard from '@cdo/apps/lib/kits/maker/boards/microBit/MicroBitBoard';
-import VirtualMBBoard from '@cdo/apps/lib/kits/maker/boards/VirtualMBBoard';
```
Comment (molly-moen) [inline apps/test/unit/lib/kits/maker/boards/MakerBoardInterfaceTestUtil.js:8]: can you add a comment to this file and the other test util file explaining that they export tests as a function? Is my understanding correct that we won't run these test files directly?

**Review by molly-moen — APPROVED** [review]
LGTM, just one suggestion on commenting!

File: apps/test/unit/lib/kits/maker/boards/MakerBoardInterfaceTestUtil.js (hunk 1 above)
Comment (fisher-alice) [reply apps/test/unit/lib/kits/maker/boards/MakerBoardInterfaceTestUtil.js:8]: Sure - will do! And yes, we won't run these test files directly.

Comment (fisher-alice) [reply apps/test/unit/lib/kits/maker/boards/MakerBoardInterfaceTestUtil.js:8]: Actually - there is already a comment about this on line 1 above. But I'll add to the other test files. Thanks!

**Pull request merged**